addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/PresenceDetection.java

```diff
@@ -326,14 +330,23 @@ public boolean performPresenceDetection(boolean waitForDetectionToFinish) {
             });
         }
 
-        // ARP ping for IPv4 addresses. Use an own executor for each network interface
+        // ARP ping for IPv4 addresses. Use single executor for Windows tool and 
+        // each own executor for each network interface for other tools
         if (interfaceNames != null) {
-            for (final String interfaceName : interfaceNames) {
+            if (arpPingMethod != ArpPingUtilEnum.ELI_FULKERSON_ARP_PING_FOR_WINDOWS) {
```

Comment:
Much better. But: You are not relying on "interfaceNames!=null" so you might want to move that case out of the if context. Please also add a comment to your case, ie "The eli fulkerson tool handles multiple interfaces itself" or similar.